### PR TITLE
Metrics for EntryLogManagerForEntryLogPerLedger.

### PR DESCRIPTION
**Pull request by reddycharan**
Descriptions of the changes in this PR:

Add counters/statsloggers for EntryLogManagerForEntryLogPerLedger.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -114,7 +114,7 @@
     <commons-configuration.version>1.10</commons-configuration.version>
     <commons-compress.version>1.15</commons-compress.version>
     <commons-lang.version>2.6</commons-lang.version>
-    <commons-lang3.version>3.3.2</commons-lang3.version>
+    <commons-lang3.version>3.6</commons-lang3.version>
```
Comment (eolivelli) [inline pom.xml]: It is worth to add in the description of the pr a line about this library upgrade, so that it will be stated in git commit log

**Comment by sijie** [comment]
@reddycharan there are a couple issues on CI

- I think you might need to merge to latest master for some checkstyle issues.
- bump the commons-lang3 requires updating the LICENSE files : http://bookkeeper.apache.org/community/licensing/
- bump the commons-lang3 seems introduce some deprecation warnings.

**Comment by reddycharan** [comment]
Hey @sijie 

> I think you might need to merge to latest master for some checkstyle issues

I ran "mvn clean apache-rat:check install spotbugs:check -DskipTests" locally, before and after rebasing to latest master and i see no issues. So not sure what is causing problem here.

> bump the commons-lang3 requires updating the LICENSE files

didn't know about it, will follow previous examples. BTW I need to update version of apache.commons.lang3 because APIs like incrementAndGet of MutableInt are added in version 3.5 https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/mutable/MutableInt.html#incrementAndGet--

> bump the commons-lang3 seems introduce some deprecation warnings

not sure what you are referring to and what should I do about it.

**Comment by sijie** [comment]
@reddycharan : please check all the ci job results.

> I ran "mvn clean apache-rat:check install spotbugs:check -DskipTests" locally, before and after rebasing to latest master and i see no issues. 

checkstyle plugin was updated on #1434 

so can you run "mvn checkstyle:check" to check?

https://builds.apache.org/job/bookkeeper_precommit_pullrequest_java8/1134/org.apache.bookkeeper$bookkeeper-server/console

> BTW I need to update version of apache.commons.lang3 because APIs like incrementAndGet of MutableInt are added in version 3.5 

Understand that. You just need to make sure LICENSE is updated.

> not sure what you are referring to and what should I do about it.

https://travis-ci.org/apache/bookkeeper/jobs/386427504

```
[WARNING] /home/travis/build/apache/bookkeeper/tests/integration/cluster/src/test/java/org/apache/bookkeeper/tests/integration/cluster/BookKeeperClusterTestBase.java:[37,32] org.apache.commons.lang3.RandomStringUtils in org.apache.commons.lang3 has been deprecated
[WARNING] /home/travis/build/apache/bookkeeper/tests/integration/cluster/src/test/java/org/apache/bookkeeper/tests/integration/stream/StreamClusterTestBase.java:[32,32] org.apache.commons.lang3.RandomStringUtils in org.apache.commons.lang3 has been deprecated
```

**Comment by reddycharan** [comment]
@jvrao fyi.

**Comment by reddycharan** [comment]
@sijie I added @SuppressWarnings tag to the classes, but still build is failing because of warning messages. https://github.com/apache/bookkeeper/pull/1460/commits/4d2ff219135fb806f5ca2848bf7ec6724a257ff2

**Comment by reddycharan** [comment]
also locally I'm doing -

mvn clean apache-rat:check install spotbugs:check -DskipTests
mvn checkstyle:check

and they are working fine for me locally

**Comment by sijie** [comment]
@reddycharan 

I don't think it failed with warnings. It seems failed on compilation. You added a field in the EntryLogger constructor in c5f8a80d4ea8b27226737bfd62657a7e807a52ac

```
[ERROR] /home/travis/build/apache/bookkeeper/bookkeeper-server/src/test/java/org/apache/bookkeeper/bookie/SlowInterleavedLedgerStorage.java:[51,13] no suitable constructor found for EntryLogger(org.apache.bookkeeper.conf.ServerConfiguration,org.apache.bookkeeper.bookie.LedgerDirsManager,org.apache.bookkeeper.bookie.EntryLogger.EntryLogListener)
    constructor org.apache.bookkeeper.bookie.EntryLogger.EntryLogger(org.apache.bookkeeper.conf.ServerConfiguration,org.apache.bookkeeper.bookie.LedgerDirsManager) is not applicable
      (actual and formal argument lists differ in length)
    constructor org.apache.bookkeeper.bookie.EntryLogger.EntryLogger(org.apache.bookkeeper.conf.ServerConfiguration,org.apache.bookkeeper.bookie.LedgerDirsManager,org.apache.bookkeeper.bookie.EntryLogger.EntryLogListener,org.apache.bookkeeper.stats.StatsLogger) is not applicable
      (actual and formal argument lists differ in length)
```

**Comment by reddycharan** [comment]
@sijie i fixed all the compile time / checkstyle issues, I guess this should be good to go.

**Comment by sijie** [comment]
run pr validation

**Comment by sijie** [comment]
actually all CI passed. the failed check is a dangling check. ignore ci to merge.

**Comment by sijie** [comment]
IGNORE CI

**Pull request closed without merging**